src/ol/style/expressions.js

```diff
@@ -21,15 +21,18 @@ import {asArray, fromString, isStringColor} from '../color.js';
  *     of bands, depending on the underlying data source and
  *     {@link import("../source/GeoTIFF.js").Options configuration}. `xOffset` and `yOffset` are optional
  *     and allow specifying pixel offsets for x and y. This is used for sampling data from neighboring pixels.
- *   * `['get', 'attributeName']` fetches a feature attribute (it will be prefixed by `a_` in the shader)
+ *   * `['get', 'attributeName', typeHint]` fetches a feature attribute (it will be prefixed by `a_` in the shader)
  *     Note: those will be taken from the attributes provided to the renderer
+ *     A type hint can optionally be specified, in case the resulting expression contains a type uncertainty which
+ *     will make it invalid. Type hints can be one of: 'string', 'color', 'number', 'boolean', 'number_array'
  *   * `['resolution']` returns the current resolution
  *   * `['time']` returns the time in seconds since the creation of the layer
- *   * `['var', 'varName']` fetches a value from the style variables, or 0 if undefined
+ *   * `['var', 'varName', typeHint]` fetches a value from the style variables; will throw an error if variable is undefined
+ *     A type hint can optionally be specified (see 'get' operator)
```

Comment:
Can we infer the type from the variable value during parsing if needed?

Reply:
I considered it but couldn't find a way to do it that didn't feel hacky (like just moving the variables object around). I'll look into it again

Reply:
Addressed: the style being parsed is now present in the parsing context, which makes sense in the general sense I think.